apps/dashboard/app/models/remote_transfer.rb

```diff
@@ -0,0 +1,157 @@
+          existing = RcloneUtil.lsf(record.dest_remote, record.to).map { |file| File.join(record.to, file) }
+          conflicts = files.values.intersection(existing)
+          record.errors.add :base, "These files already exist: #{conflicts.join(', ')}" if conflicts.present?
+        rescue RcloneError => e
+          if e.exitstatus != 3
+            # Rclone will return status 3 if directory doesn't exist (ok), other errors are unexpected
+            record.errors.add :base, "Error checking existing files in destination: #{e}"
+          end
+        end
+      end
+    end
+  end
+
+  attr_accessor :src_remote, :dest_remote, :filesizes, :transferred
+
+  class << self
+    def transfers
+      # all transfers stored in the Transfer class
+      Transfer.transfers
+    end
+
+    def build(action:, files:, src_remote:, dest_remote:)
+      if files.is_a?(Array)
+        # rm action will want to provide an array of files
+        # so if it is an Array we convert it to a hash:
+        #
+        # convert [a1, a2, a3] to {a1 => nil, a2 => nil, a3 => nil}
+        files = Hash[files.map { |f| [f, nil] }]
+      end
+
+      self.new(action: action, files: files, src_remote: src_remote, dest_remote: dest_remote)
+    end
+  end
+
+  # total number of bytes
+  def steps
+    return @steps if @steps
+
+    @filesizes = {}
+
+    # Get info from `rclone size` (will not work on Google Drive and Google Photos)
+    total_size = files.keys.map do |file|
+      size = RcloneUtil.size(src_remote, file).fetch('bytes', 0)
+      @filesizes[file] = { :size => size, :transferred => 0 }
+      size
+    end.sum
+
+    @steps = total_size
+  end
+
+  def command_str
+    ''
+  end
+
+  def increment_transferred(amount)
+    @transferred = transferred.to_i + amount
+  end
+
+  # Updates the total progress with progress of one file since last update
+  def update_progress(file, percent_done)
+    file_info = filesizes[file]
+
+    current_bytes = (percent_done * file_info[:size]) / 100
+    since_last = current_bytes - file_info[:transferred]
+    filesizes[file][:transferred] = current_bytes
+
+    increment_transferred(since_last)
+    update_percent(transferred)
+  end
+
+  def perform
+    self.status = OodCore::Job::Status.new(state: :running)
+    self.started_at = Time.now.to_i
+
+    # Store info about sizes of files to transfer for tracking progress
+    steps
+
+    # Transfer each file/directory indiviually
+    files.each do |src, dst|
+      if action == 'mv'
+        RcloneUtil.moveto_with_progress(src_remote, dest_remote, src, dst) do |p|
+          update_progress(src, p)
+        end
+      elsif action == 'cp'
+        RcloneUtil.copyto_with_progress(src_remote, dest_remote, src, dst) do |p|
+          update_progress(src, p)
+        end
+      elsif action == 'rm'
+        RcloneUtil.remove_with_progress(src_remote, src) do |p|
+          update_progress(src, p)
+        end
+      else
+        raise "Unknown action: #{action.inspect}"
```

Comment:
```suggestion
        raise StandardError, "Unknown action: #{action.inspect}"
```

`raise` defaults to RuntimeError which could be harder to catch than a `StandardError`.

Reply:
Fixed.